Makefile

```diff
@@ -93,6 +93,11 @@ manifests: controller-gen ## Generate WebhookConfiguration, ClusterRole and Cust
 generate: controller-gen ## Generate code containing DeepCopy, DeepCopyInto, and DeepCopyObject method implementations.
 	$(CONTROLLER_GEN) object:headerFile="hack/boilerplate.go.txt" paths="./..."
 
+.PHONY: mock
+mock:
+	$(RM) test/k8sclient/mock_client.go
+	mockgen -destination=test/k8sclient/mock_client.go -package=k8sclient "github.com/armadaproject/armada-operator/test/k8sclient" Client
```

Comment:
Probably should have a target that pulls down mockgen.  You can look at the bottom of the Makefile for a template.  

Not sure what RM is here also.

Reply:
`$(RM)` is a standard GNU Make alias to remove a file - on Unix/Linux/macOS, it expands to `rm -f`, and if running on Windows (god forbid), I believe it expands to `DEL` or somesuch.

I'll add a step to fetch mockgen.

Reply:
Today I learned!

Reply:
Good suggestion - mockgen command fetch added in commit d3971fb.